spec.md

```diff
@@ -413,6 +419,22 @@ message CreateVolumeResponse {
   }
 }
 
+message Principal {
+  // describes attributes for defining the principal identity of 
+  // a volume.
+
+  // This field is OPTIONAL. If specified it contains the CO namespace
+  // identity in which the the following identity information (user /
+  // group) exists.
+  string namespace = 1;
```

Comment:
Sorry for the very late response, but I am a bit unclear as to how this change addresses the case that govint raised.  Say I am a plugin developer, and my storage solution has some proprietary IdP or authentication scheme that is not known to the CO.  The end user has to issue some sort of command to the CO that associates the storage instance to the container, and presumably provides credentials for the storage in some kind of data blob.  The CO stores that data blob somewhere and then calls controller publish & node publish.  So far so good.

What I don't get is, why would the Principal we pass to the plugin be a *CO* user and group?  What would a plugin author do with that information that isn't CO specific?  Why not instead just leave the credentials as a set of key/value pairs that are opaque to the CO and interpreted by the plugin?  That way if the plugin requires a ceph keyring or kerberos keytab file or what have you, the CO can just pass it through.

Or is the idea that CO proprietary credentials blocks can be nested into `PublishVolumeInfo` and we just want the `Principal` for plugins that use the CO as an IdP?